contrib/babelfishpg_tsql/sql/ownership.sql

```diff
@@ -14,6 +14,18 @@ CREATE TABLE sys.babelfish_sysdatabases (
 
 GRANT SELECT on sys.babelfish_sysdatabases TO PUBLIC;
 
+-- BABELFISH_FUNC_DEFAULT
+CREATE TABLE sys.babelfish_function_ext (
+	nspname NAME NOT NULL,
+	funcname NAME NOT NULL,
+	funcsignature TEXT NOT NULL COLLATE "C",
+	default_positions TEXT COLLATE "C",
+	PRIMARY KEY(nspname, funcsignature)
```

Comment:
why funcname is not part of the primary key ?  
My understanding is that you could implement drop / alter procedure and functions only when PK is identical to base catalog table? How do you handle drop/ / alter case. 
And is there a reason to exclude stored procedure from the beginning ?

Reply:
Since we are introducing catalog for pg_proc, there was a need to store users' original input name too. Please check with Rob and leave a column for it even if not filling it at the moment.

Reply:
> why funcname is not part of the primary key ? My understanding is that you could implement drop / alter procedure and functions only when PK is identical to base catalog table? How do you handle drop/ / alter case. And is there a reason to exclude stored procedure from the beginning ?

`funcsignature` actually consists of function name + arguments (e.g.: `f1(int, varchar)`) so `funcname` is not needed in PK. `schema name + function signature` can actually uniquely identify a function. I added a `funcname` column just to for the future reference, in case we need it. I am using `schema name + function signature` only for `DROP` also but currently I am not handling `ALTER` currently since alter function is not supported for functions in babel.
Also, can you clarify what are `DDL behavior` tests or point me to some existing DDL behavior tests present in our framework?

Reply:
Yes you would need to add some rules in integrity checker (in catalog.c)
I can think of several rules like
For each entry in babelfish_function_ext, 
1. nspname must exist in babelfish_namespace_ext
2. funcname must exist in pg_proc

It's developer's responsibility to list out all possible rules, i.e. the connection between babelfish_function_ext and other catalogs, especially PG catalogs, and add them in the checker.
Feel free to ask me if you have any other questions.

Reply:
Added above integrity checks